### ISSUE TITLE
Is it useful with kernel 5.18 (new rng blake2 output - entropy_avail always equal to 256)?

### ISSUE THREAD

**Issue by pwasiewi** [nhorman/rng-tools#181]
In kernel 5.18 a new rng method based on BLAKE2 was introduced.  Do rng-tools cooperate with RNG improvements or are just obsolete?
The same question here (no answer):
https://www.reddit.com/r/linuxquestions/comments/tq732u/kernel_517_rngtools_still_useful/

**Comment by nhorman** [comment]
thats a question with alot of answers.  The recent improvements in kernel entropy generation do mitigate the need to run rngd on some systems.  That said:
1) kernel 5.18 is _new_, just because the head of upstream has some new bell/whistle doesn't mean that every linux user in the world uses it.  Many linux installations at currently deployment levels will exist for years/decades, and for them rngd is still needed
2) Some users want entropy from specific sources, not whatever cprng the kernel provides. For them, the ability to run a daemon that can gather entropy from a source that is under their control and feed it to the kernel is beneficial
3) rng-tools is more than a daemon for kernel entropy population.  Its useful for entropy gathering for a wide variety of uses

So in short, not, rng-tools is not obsolete.  It may just not be used as frequently for kernel entropy generation in the future
